crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs

```diff
@@ -0,0 +1,327 @@
+    for potential in split {
+        if let Some(first_char) = potential.chars().nth(indentation) {
+            if !first_char.is_whitespace() {
+                let entry = potential[indentation..].trim().to_string();
+                entries.push(entry);
+            }
+        }
+    }
+    entries
+}
+
+// Parse body
+#[derive(Debug)]
+struct Entry {
+    id: String,
+    range: TextRange,
+}
+
+struct BodyEntries {
+    raised_exceptions: Vec<Entry>,
+}
+
+struct BodyVisitor<'a> {
+    raised_exceptions: Vec<Entry>,
+    semantic: &'a SemanticModel<'a>,
+}
+
+impl<'a> BodyVisitor<'a> {
+    fn new(semantic: &'a SemanticModel<'a>) -> Self {
+        Self {
+            raised_exceptions: Vec::new(),
+            semantic,
+        }
+    }
+
+    fn finish(self) -> BodyEntries {
+        BodyEntries {
+            raised_exceptions: self.raised_exceptions,
+        }
+    }
+}
+
+impl Visitor<'_> for BodyVisitor<'_> {
+    fn visit_stmt(&mut self, stmt: &Stmt) {
+        if let Stmt::Raise(ast::StmtRaise { exc: Some(exc), .. }) = stmt {
+            match exc.as_ref() {
+                Expr::Name(ast::ExprName { id, range, .. }) => {
+                    // SemanticModel will resolve qualified_name for local Class definitions,
+                    // or imported definitions, but not variables which we want to ignore.
+                    if self.semantic.resolve_qualified_name(exc.as_ref()).is_some() {
+                        self.raised_exceptions.push(Entry {
+                            id: id.to_string(),
+                            range: *range,
+                        });
+                    }
+                }
+                Expr::Call(ast::ExprCall { func, range, .. }) => {
+                    if let Expr::Name(ast::ExprName { id, .. }) = func.as_ref() {
+                        // SemanticModel will resolve qualified_name for local Class definitions,
+                        // or imported definitions, but not variables which we want to ignore.
+                        if self
+                            .semantic
+                            .resolve_qualified_name(func.as_ref())
+                            .is_some()
+                        {
+                            self.raised_exceptions.push(Entry {
+                                id: id.to_string(),
+                                range: *range,
+                            });
+                        }
+                    }
+                }
+                _ => {}
+            };
+        }
+        visitor::walk_stmt(self, stmt);
+    }
+}
+
+/// DOC501, DOC502
+pub(crate) fn check_docstring(
+    checker: &mut Checker,
+    definition: &Definition,
+    section_contexts: &SectionContexts,
+    convention: Option<&Convention>,
+) {
+    let Definition::Member(member) = definition else {
+        return;
+    };
+
+    if matches!(
+        member.kind,
+        MemberKind::Class(_) | MemberKind::NestedClass(_)
```

Comment:
This implies function-only, right?

Reply:
Yes. At least for now since exceptions can't be raised by classes AFAIK.